examples/multi_gpu/distributed_sampling_xpu.py

```diff
@@ -127,20 +139,23 @@ def run(rank: int, world_size: int, dataset: PygNodePropPredDataset):
     optimizer = torch.optim.Adam(model.parameters(), lr=0.001)
 
     for epoch in range(1, 21):
+        epoch_stats = []
         model.train()
         for batch in train_loader:
-            optimizer.zero_grad()
-            out = model(batch.x,
-                        batch.edge_index.to(device))[:batch.batch_size]
-            loss = F.cross_entropy(out, batch.y[:batch.batch_size].squeeze())
-            loss.backward()
-            optimizer.step()
+            batch = batch.to(device)
+            loss, stats = train_step(model, optimizer, batch.x,
+                                     batch.edge_index, batch.y,
+                                     batch.batch_size)
```

Comment:
```suggestion
            loss, stats = train_step(model, optimizer, batch)
```

Reply:
This cannot be done, as in the current design `profileit` decorator detects device id based on input tensors.